### PR TITLE
CRM-19302 - Fix Copying of File Custom Fields on Event Copy

### PR DESCRIPTION
**Pull request by MiyaNoctem**
Fixed bug! Created new static method that copies all custom fields from old event to new event.  Files are handled by copying them and having separate records in civicrm_file table, not just custom fields table.

---
- [CRM-19302: Event copy - file type custom data not being copied properly ](https://issues.civicrm.org/jira/browse/CRM-19302)

### CODE REVIEW

File: CRM/Event/BAO/Event.php [hunk 1]
```diff
@@ -96,9 +96,13 @@ public static function add(&$params) {
     }
 
     // CRM-16189
-    if (!empty($params['financial_type_id'])) {
-      CRM_Financial_BAO_FinancialAccount::validateFinancialType($params['financial_type_id']);
+    $isError = CRM_Financial_BAO_FinancialAccount::validateFinancialType(
+      CRM_Utils_Array::value('financial_type_id', $params)
+    );
+    if ($isError) {
+      throw new CRM_Core_Exception(ts('Deferred revenue account is not configured for selected financial type. Please have an administrator set up the deferred revenue account at Administer > CiviContribute > Financial Accounts, then configure it for financial types at Administer > CiviContribution > Financial Types, Accounts'));
```
Comment (seamuslee001) [inline CRM/Event/BAO/Event.php]: won't the validateFinancialType already throw and exception? also not sure why your removing the !empty that is around it.

Comment (MiyaNoctem) [inline CRM/Event/BAO/Event.php]: Actually, that's not part of the code I fixed, but it was on the source I worked on... seems to be related to bug report 16189, not really sure how that got there, though...

**Comment by colemanw** [comment]
@civicrm-builder add to whitelist

File: CRM/Event/BAO/Event.php (hunk 1 above)
Comment (colemanw) [inline CRM/Event/BAO/Event.php]: @MiyaNoctem could you please update this PR with only the changes you meant to make?

**Review by colemanw — CHANGES_REQUESTED** [review]

File: CRM/Event/BAO/Event.php [hunk 2]
```diff
@@ -96,6 +96,10 @@ public static function add(&$params) {
     }
 
     // CRM-16189
+    if (!empty($params['financial_type_id'])) {
+      CRM_Financial_BAO_FinancialAccount::validateFinancialType($params['financial_type_id']);
+    }
+   
     $isError = CRM_Financial_BAO_FinancialAccount::validateFinancialType(
       CRM_Utils_Array::value('financial_type_id', $params)
     );
```
Comment (colemanw) [inline CRM/Event/BAO/Event.php]: This block seems to do the same thing as the above?

**Comment by Edzelopez** [comment]
Tested patch on 4.7, works fine.

**Comment by monishdeb** [comment]
Jenkin test this please

**Comment by monishdeb** [comment]
Closed in favor of https://github.com/civicrm/civicrm-core/pull/9407

**Pull request closed without merging**